timeseries/src/main/java/com/hurence/logisland/timeseries/sax/SaxAnalyzer.java

```diff
@@ -0,0 +1,75 @@
+package com.hurence.logisland.timeseries.sax;
+
+import com.sun.javafx.collections.MappingChange;
+
+import java.util.*;
+import java.util.stream.Collectors;
+
+import static java.util.stream.Collectors.*;
+
+
+public class SaxAnalyzer {
+
+    public static Map<Integer, Integer> countOccurrence(int[] numbersToProcess) {
+        int[] possibleNumbers = new int[30];
+        Map<Integer, Integer> result = new HashMap<Integer, Integer>();
+
+        for (int i = 0; i < numbersToProcess.length; ++i) {
+            possibleNumbers[numbersToProcess[i]] = possibleNumbers[numbersToProcess[i]] + 1;
+            result.put(numbersToProcess[i], possibleNumbers[numbersToProcess[i]]);
+        }
+
+        return result;
+    }
+    public static List<Integer> saxThreshold(String str){
```

Comment:
saxThresholds must take a **threshold** int parameter

Reply:
that's the idea at first. But it won't be functional in the case of "saxsaxsaxsaxh" where the distance between the last 'x' and 'h' is less then the distance between ('s','a') and the distance between ('a','x'). In the case of one int threshold we will get anomaly in every position. the solution was to get a list of anomaly distances and search for them in the sax string.